src/test/java/orlov/programming/spring_core_gym/dao/impl/training/TrainingDAOTest.java

```diff
@@ -0,0 +1,142 @@
+package orlov.programming.spring_core_gym.dao.impl.training;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import orlov.programming.spring_core_gym.dao.DAO;
+import orlov.programming.spring_core_gym.model.training.Training;
+import orlov.programming.spring_core_gym.model.user.Trainee;
+import orlov.programming.spring_core_gym.model.user.Trainer;
+import orlov.programming.spring_core_gym.storage.impl.TrainingStorage;
+
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TrainingDAOTest {
+    private static final String TRAINING_NULL_ERROR = "Training can't be null";
+    private static final String TRAINING_NOT_FOUND_ERROR_MESSAGE = "Training is not found for ";
+    private static final String TRAINEE_NOT_FOUND_ERROR_MESSAGE = "Trainee is not found for ";
+    private static final String TRAINER_NOT_FOUND_ERROR_MESSAGE = "Trainer is not found for ";
+
+    @Mock
+    private DAO<Trainee> traineeDAO;
+
+    @Mock
+    private DAO<Trainer> trainerDAO;
+
+    @Mock
+    private TrainingStorage storage;
```

Comment:
My recommendation to use one class storage per all daos, it will be simulation of DB